Document how IPv6-only works in Docker Desktop (#20918)
docker/docs: content/manuals/desktop/networking.md

See the [run command](/reference/cli/docker/container/run.md) for more details on
publish options used with `docker run`.

> ### IPv4-only, Dual-stack and IPv6-only (beta)

Reviewer: What is the default?

Author: I added a note about the default mode in the first paragraph.

> [!NOTE]
> 
> This feature is under active development and the exact semantics of all three
> options might change in future releases.

Starting with v4.35, Docker Desktop has new options to configure how the VM behaves and the default
settings applied to Docker networks.

> With IPv4-only, the DNS resolver embedded in the VM will only serve A (= IPv4) records and suppress
> any AAAA (= IPv6) records. This might be useful if some of your containerized software try to

Reviewer: ```suggestion
In IPv4-only mode, the DNS resolver embedded in the VM only serves A (IPv4) records and suppresses
any AAAA (IPv6) records. This might be useful if some of your containerized software try to
```

connect to IPv6 addresses but your host doesn't have proper IPv6 connectivity.

> With Dual-stack, both A and AAAA records are served by the DNS resolver. Docker networks are

Reviewer: ```suggestion
With Dual-stack IPv4 and IPv6, both A and AAAA records are served by the DNS resolver. Docker networks are
```

Author: I'd prefer to stick with the option name shown in the UI.

> created with an IPv4 subnet assigned, but no IPv6 subnet. You need to pass in `--ipv6` to

Reviewer: ```suggestion
created with an IPv4 subnet assigned, but no IPv6 subnet. You must pass the `--ipv6` flag to
```

[`docker network create`](https://docs.docker.com/reference/cli/docker/network/create/#options),
or put [`enable_ipv6: true`](https://docs.docker.com/reference/compose-file/networks/#enable_ipv6)
in network declarations in Compose files.

> With IPv6-only, the DNS resolver serves only AAAA records, and the Engine is configured to assign

Reviewer: ```suggestion
In IPv6-only mode, the DNS resolver serves only AAAA records, and Docker Engine assigns
```

IPv6 subnets (but no IPv4) by default to your networks. This might be useful if you run Docker
Desktop in an environment where IPv4 is getting phased out but has not been entirely shut down yet.

> For more details about IPv6 in Docker Engine refer to [https://docs.docker.com/engine/daemon/ipv6/](https://docs.docker.com/engine/daemon/ipv6/).

Reviewer: ```suggestion
For more details about IPv6 in Docker Engine refer to [Use IPv6 networking](/manuals/engine/daemon/ipv6.md).
```

See the [run command](/reference/cli/docker/container/run.md) for more details on
publish options used with `docker run`.

### IPv4-only, Dual-stack and IPv6-only

{{% experimental title="Beta" %}}
This feature is currently in [Beta](../release-lifecycle.md#beta).  

The exact semantics of all three options might change in future releases.

> {{% /experimental %}}

Reviewer 2: ```suggestion

Note: IPv6 is not available for Docker Desktop on Linux.
{{% /experimental %}}
```
(due to the limitation with qemu + gvisor)
